text/0000-landmarks.md

```diff
@@ -0,0 +1,99 @@
+# Landmark observations
+Add landmarks support to Cartographer.
+
+## Summary
+[summary]: #summary
+Cartographer will be able to work with a new type of sensor data that contains IDs and locations of landmarks.
+
+## Motivation
+[motivation]: #motivation
+Landmarks translation/orientation can improve SLAM quality.
+Landmark visualizations can be useful for inspection.
+
+## Approach
+[approach]: #approach
+Landmarks are sparse features of geometry that we want to map.
+Unlike dense range data, landmarks are assumed to be sparse in comparison.
+In order to support using them, current SLAM approach has to be generalized.
+
+### Sensor
+Each landmark observation contains all landmarks for which a relative position is currently observed.
+This can be an empty list.
+
+```
+struct Landmark {
```

Comment:
I would rename this to `LandmarkObservation` to make the difference to `LandmarkNode` more explicit.